R/lockbox.R

```diff
@@ -63,7 +64,7 @@ lockbox.list <- function(lock, env) {
     align(lock[mismatches])
 
     ## And re-build our search path. Do so in the reverse order of dependencies.
-    rebuild(rev(lock))
+    rebuild(c(lock[vapply(lock, `[[`, logical(1), "is_dependency_package")], original_lock))
```

Comment:
So everything in the 'lock' list that is a dependency package should not have been in the original lockfile, and the final lock list doesn't have any duplicates, so the dependencies in lock and the packages in the original locked list should be disjoint (unless my eyeballs are being dumb and I'm missing something...)